src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -172,6 +172,11 @@ private void CancellationCallback()
 
             internal void Start()
             {
+                // Make sure _fullPath doesn't contain a link or alias
+                // since the OS will give back the actual, non link'd or alias'd paths
+                int result = Interop.Sys.RealPath(_fullDirectory, _fullDirectory);
```

Comment:
Is the issue this PR is solving OS X specific?  Does it also exist on Linux, or does inotify differ in how it handles such things?